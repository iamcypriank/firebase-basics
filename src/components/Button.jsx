export default function Button({ type="button", variant, children , onClick, ariaLabel }){

    let style = '';
    if(variant=='primary') style='px-8 py-2 bg-blue-600 text-white font-medium rounded-md';
    if(variant=='secondary') style='box-border px-8 py-2 border-2 border-blue-600  text-black font-medium rounded-md';
    if(variant =='thirdPartyLoginButton') style = 'flex justify-center py-2 border-2 border-gray-300 rounded-full hover:bg-gray-100 transition';

    return ( 
    <button
    aria-label={ariaLabel}
    className={`${style}`} 
    onClick={(e)=>{
        if(type=='button'){
            e.preventDefault();
            onClick();
        }
    }} >
        {children}
    </button> )
}