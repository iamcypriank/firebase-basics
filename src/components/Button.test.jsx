import  { describe, it, expect, vi } from "vitest";
import { screen ,render } from "@testing-library/react";
import Button from "./Button";
import userEvent from "@testing-library/user-event";
describe('Button Component Test Cases : ',()=>{

    it('button renders',()=>{
        render(<Button/>)
        expect(screen.getByRole('button')).toBeInTheDocument();
    })

    it('button renders correct text',()=>{
        render(<Button>login</Button>)
        expect(screen.getByRole('button',{ name : 'login' })).toBeInTheDocument();
    })


    it('button should call onlick function',async ()=>{ 
        const onClick= vi.fn();
        const user = userEvent.setup();
        render(<Button onClick={onClick}  >login</Button>)
        const button = screen.getByRole('button',{ name : 'login' });
        
        await user.click(button);

        expect(onClick).toHaveBeenCalled();
    })


    it('button should not call onlick function',async ()=>{ 
        const onClick= vi.fn();
        const user = userEvent.setup();
        render(<Button type="submit" onClick={onClick}  >login</Button>)
        const button = screen.getByRole('button',{ name : 'login' });
        
        await user.click(button);

        expect(onClick).not.toHaveBeenCalled();
    })

})