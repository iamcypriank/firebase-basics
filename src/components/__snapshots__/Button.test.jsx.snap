// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component Test Cases :  > button is rendering 1`] = `
<div>
  <button
    class=""
  />
</div>
`;
